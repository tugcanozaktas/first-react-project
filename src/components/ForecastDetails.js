/* eslint-disable react/no-unused-prop-types */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  return (
    <div className="forecast-details">
      <div>{moment(date).format("ddd Do MMM")}</div>
      <div>Max Temperature: {temperature.max}&deg;C</div>
      <div>Max Temperature: {temperature.min}&deg;C</div>
      <div>Humidity: {humidity}%</div>
      <div>
        Wind: {wind.speed}mph {wind.direction}
      </div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    humidity: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default ForecastDetails;
