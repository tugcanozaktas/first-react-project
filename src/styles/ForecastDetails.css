.forecast-details{
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 20px;
    border-radius: 10px;

    background-color: #7388b9ad;
    color: #222c41;
}

.forecast-details div{
    text-align: left;
    display: inline-block;
    width: 200px;
}
#date{
    font-size: 20px;
    text-decoration: underline;
    text-align: center;
    font-weight: bold;
}