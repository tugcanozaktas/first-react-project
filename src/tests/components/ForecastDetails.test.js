import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1525046400000,
    humidity: 25,
    wind: {
      speed: 25,
      direction: "ne",
    },
    temperature: {
      max: 22,
      min: 12,
    },
  };
  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment).toMatchSnapshot();
  });
});
