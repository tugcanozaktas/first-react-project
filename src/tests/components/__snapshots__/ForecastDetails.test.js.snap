// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `[Function]`;
